src/stats.c

```diff
@@ -41,14 +41,45 @@
 
 #include <assert.h>
 
+struct StatsState {
+/****************************************************************************
+**
+*V  ReturnObjStat . . . . . . . . . . . . . . . .  result of return-statement
+**
+**  'ReturnObjStat'  is   the result of the   return-statement  that was last
+**  executed.  It is set  in  'ExecReturnObj' and  used in the  handlers that
+**  interpret functions.
+*/
+Obj ReturnObjStat;
+UInt (**CurrExecStatFuncs)(Stat);
+};
+
+static ModuleStateOffset StatsStateOffset = -1;
+extern inline struct StatsState * StatsState(void)
+{
+    return (struct StatsState *)StateSlotsAtOffset(StatsStateOffset);
+}
+
+extern inline void SetCurrExecStatFuncs(UInt (**newExecStatFuncs)(Stat))
+{
+    StatsState()->CurrExecStatFuncs = newExecStatFuncs;
+}
 
 inline UInt EXEC_STAT(Stat stat)
 {
     UInt tnum = TNUM_STAT(stat);
     SET_BRK_CALL_TO(stat);
-    return (*STATE(CurrExecStatFuncs)[ tnum ]) ( stat );
+    return (*StatsState()->CurrExecStatFuncs[ tnum ]) ( stat );
 }
 
+inline Obj EXEC_STAT_RETURN_OBJ(Stat stat)
```

Comment:
Note that in all calls, `stat == OFFSET_FIRST_STAT`. So you could also remove the argument, and call this something like `EXEC_CURR_FUNC` or so... I think that would make for a nice documentation comment, too.